mantidproject/mantid
# GSASIIRefineFitPeaks make sigma and gamma refinable

The sigma and gamma profile coefficients were added to `GSASIIRefineFitPeaks`. Flags for whether to refine them or not were added, and their values were added as output properties of the algorithm.

**To test:**

**Make sure GSASIIRefineFitPeaks system test passes locally - this is not currently run on the build server**

You'll need GSAS-II installed. Install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS).

Run the following script (replace the marked variables with your GSAS-II installation directory and the name of a new GSAS-II project file, which will be created by the algorithm)

```
import os

#replace this
path_to_gsas = "/path/to/directory/containing/GSASIIscriptable.py"

data_dir = r"\\olympic\Babylon5\Public\JoeRamsay\GSASIIRefineFitPeaks"

ws_path = os.path.join(data_dir, "ENGINX_280625_focused_bank_1.nxs")
spectrum = Load(Filename=ws_path, OutputWorkspace="ENGINX_280625_1")

#replace this
gsas_proj_file = "/somewhere/local/SigmaGammaTest.gpx"

phase_file = os.path.join(data_dir, "FE_ALPHA.cif")
iparams_file = os.path.join(data_dir, "template_ENGINX_241391_236516_North_bank.prm")

_, _, sig, gam, _ = GSASIIRefineFitPeaks(PathToGSASII=path_to_gsas, 
                     InputWorkspace=spectrum,
                     SaveGSASIIProjectFile=gsas_proj_file,
                     PhaseInfoFile=phase_file,
                     InstrumentFile=iparams_file,
                     RefinementMethod="Pawley refinement",
                     RefineSigma=True,
                     RefineGamma=True)
                     
print("Sigma: {}".format(sig))
print("Gamma: {}".format(gam))
```

Sigma should be `-188,8`, gamma should be `37.2`.

Fixes #21575 

**Release Notes**
Doesn't need to be in the release notes, as it was possibly in the last version of `GSASIIRefineFitPeaks`, and the update to the algorithm is already in there

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
